### ISSUE TITLE
Analyzer constantly crashing

### ISSUE THREAD

**Issue by filipproch** [dart-lang/sdk#39626]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA IC-193.5233.102`
- `2.7.0-edge.867a6e0e5288fbc177e4e145287438aebaaa7034`
- `IC-193.5233.102, JRE 11.0.4+10-b520.11x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.15.1, screens 1680x1050  Retina`

## Exception
```
Dart analysis server, SDK version 2.7.0-edge.867a6e0e5288fbc177e4e145287438aebaaa7034, server version 1.27.4, FATAL error: RangeError (index): Invalid value: Valid value range is empty: 0
#0      List.[] (dart:core-patch/growable_array.dart:149:60)
#1      ElementWalker.getVariable (package:analyzer/src/generated/declaration_resolver.dart:163:46)
#2      ResolutionVisitor.visitVariableDeclaration (package:analyzer/src/dart/resolver/resolution_visitor.dart:936:32)
#3      VariableDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:10350:49)
#4      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7511:20)
#5      VariableDeclarationListImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:10452:16)
#6      ResolutionVisitor.visitVariableDeclarationList (package:analyzer/src/dart/resolver/resolution_visitor.dart:972:10)
#7      VariableDeclarationListImpl.accept (package:analyzer/src/dart/ast/ast.dart:10446:15)
#8      FieldDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:3987:17)
#9      RecursiveAstVisitor.visitFieldDeclaration (package:analyzer/dart/ast/visitor.dart:857:10)
#10     ResolutionVisitor.visitFieldDeclaration (package:analyzer/src/dart/resolver/resolution_visitor.dart:429:11)
#11     FieldDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:3982:49)
#12     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7511:20)
#13     ResolutionVisitor.visitClassDeclaration.<anonymous closure>.<anonymous closure> (package:analyzer/src/dart/resolver/resolution_visitor.dart:234:22)
#14     ResolutionVisitor._withNameScope (package:analyzer/src/dart/resolver/resolution_visitor.dart:1226:8)
#15     ResolutionVisitor.visitClassDeclaration.<anonymous closure> (package:analyzer/src/dart/resolver/resolution_visitor.dart:213:7)
#16     ResolutionVisitor._withElementWalker (package:analyzer/src/dart/resolver/resolution_visitor.dart:1215:8)
#17     ResolutionVisitor.visitClassDeclaration (package:analyzer/src/dart/resolver/resolution_visitor.dart:212:5)
#18     ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1522:49)
#19     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7511:20)
#20     CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2098:21)
#21     RecursiveAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:725:10)
#22     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2091:49)
#23     LibraryAnalyzer._resolveFile (package:analyzer/src/dart/analysis/library_analyzer.dart:675:10)
#24     LibraryAnalyzer.analyzeSync.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:154:7)
#25     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:379:8)
#26     LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:153:11)
#27     LibraryAnalyzer.analyze.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:116:14)
#28     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#29     LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:115:43)
#30     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1275:63)
#31     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:34:15)
#32     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1251:20)
#33     AnalysisDriver._computeIndex (package:analyzer/src/dart/analysis/driver.dart:1317:37)
#34     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:974:39)
<asynchronous suspension>
#35     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:1940:24)
<asynchronous suspension>
#36     AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:1874:5)
#37     new AnalysisServer (package:analysis_server/src/analysis_server.dart:223:29)
#38     SocketServer.createAnalysisServer (package:analysis_server/src/socket_server.dart:89:26)
#39     StdioAnalysisServer.serveStdio (package:analysis_server/src/server/stdio_server.dart:41:18)
#40     Driver.startAnalysisServer.<anonymous closure> (package:analysis_server/src/server/driver.dart:619:21)
#41     _rootRun (dart:async/zone.dart:1126:13)
#42     _CustomZone.run (dart:async/zone.dart:1023:19)
#43     _runZoned (dart:async/zone.dart:1518:10)
#44     runZoned (dart:async/zone.dart:1465:12)
#45     Driver._captureExceptions (package:analysis_server/src/server/driver.dart:735:12)
#46     Driver.startAnalysisServer (package:analysis_server/src/server/driver.dart:617:7)
#47     Driver.start (package:analysis_server/src/server/driver.dart:515:7)
#48     main (file:///b/s/w/ir/cache/builder/src/third_party/dart/pkg/analysis_server/bin/server.dart:12:11)
#49     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:45:6)
#50     main (file:///b/s/w/ir/cache/builder/src/third_party/dart/pkg/analysis_server/bin/server.dart:10:10)
#51     _start...
```

For additional log information, please append the contents of
file:///private/var/folders/w3/plmd8l2s2fx_3sh395mw5nr00000gn/T/report.txt.

**Comment by srawlins** [comment]
Thanks for reporting this issue! This looks like a duplicate of https://github.com/dart-lang/sdk/issues/38953 which should be fixed in 2.7.0 final. Please re-open this issue if you see this problem again after upgrading Dart / Flutter to Dart 2.7.0 final.

**Comment by srawlins** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/38953
